### PR TITLE
feat: add `discord_messages` table and persistence APIs

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
This adds a new `discord_messages` table and extends the persistence
APIs such that `MessagesByID` and `MessageByID` will return user
messages that include their discord message payload.

It also adds APIs to save individual discord messages.

### CODE REVIEW

**Comment by 0x-r4bbit** [comment]
The tests for this one keep failing but it doesn't seem to be related to the changes of this PR..

File: protocol/message_persistence.go [hunk 1]
```diff
@@ -1758,6 +1900,65 @@ func (db sqlitePersistence) SaveDiscordMessageAuthor(author *protobuf.DiscordMes
...
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	query := "INSERT INTO discord_messages(id, author_id, type, timestamp, timestamp_edited, content, reference_message_id, reference_channel_id, reference_guild_id) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)"
+	stmt, err := tx.Prepare(query)
+	if err != nil {
+		return
+	}
+
```
Comment (richard-ramos) [inline protocol/message_persistence.go:1944]:
```suggestion
defer stmt.Close()
```

File: protocol/message_persistence.go [hunk 2]
```diff
@@ -1758,6 +1900,65 @@ func (db sqlitePersistence) SaveDiscordMessageAuthor(author *protobuf.DiscordMes
 	return
 }
 
+func (db sqlitePersistence) SaveDiscordMessage(message *protobuf.DiscordMessage) (err error) {
+	query := "INSERT INTO discord_messages(id,type,timestamp,timestamp_edited,content,author_id, reference_message_id, reference_channel_id, reference_guild_id) VALUES (?,?,?,?,?,?,?,?,?)"
+	stmt, err := db.db.Prepare(query)
+	if err != nil {
+		return
+	}
+	_, err = stmt.Exec(
```
Comment (richard-ramos) [inline protocol/message_persistence.go:1910]:
```suggestion
	defer stmt.Close()
	_, err = stmt.Exec(
```

**Review by richard-ramos — APPROVED** [review]
Good job. just left some small suggestions for closing prepared statements once we're done with them

**Comment by 0x-r4bbit** [comment]
Paging my friendly neighborhood @qoqobolo 
This is ready for client testing :)

@richard-ramos, thank you!

**Pull request merged**